src/fallback.rs

```diff
@@ -0,0 +1,54 @@
+use cfg_if::cfg_if;
+
+cfg_if! {
+if #[cfg(feature = "ssr")] {
+  use axum::{
+    body::{boxed, Body, BoxBody},
+    extract::Extension,
+    response::IntoResponse,
+    http::{Request, Response, StatusCode, Uri},
+};
+use axum::response::Response as AxumResponse;
+use tower::ServiceExt;
+use tower_http::services::ServeDir;
+use std::sync::Arc;
+use leptos::{LeptosOptions, view};
+use crate::app::{App, AppProps};
+
+}}
+
+#[cfg(feature = "ssr")]
+pub async fn file_and_error_handler(
+    uri: Uri,
+    Extension(options): Extension<Arc<LeptosOptions>>,
+    req: Request<Body>,
+) -> AxumResponse {
+    let options = &*options;
+    let root = options.site_root.clone();
+    let res = get_static_file(uri.clone(), &root).await.unwrap();
+
+    if res.status() == StatusCode::OK {
+        res.into_response()
+    } else {
+        let handler =
+            leptos_axum::render_app_to_stream(options.to_owned(), move |cx| view! { cx, <App/> });
+        handler(req).await.into_response()
+    }
+}
+#[cfg(feature = "ssr")]
+async fn get_static_file(uri: Uri, root: &str) -> Result<Response<BoxBody>, (StatusCode, String)> {
+    let req = Request::builder()
+        .uri(uri.clone())
+        .body(Body::empty())
+        .unwrap();
+    let root_path = format!("{root}");
```

Comment:
Q. Why is this needed?

Reply:
not sure, I just copied from axum example